pkg/operation/botanist/secrets.go

```diff
@@ -244,6 +251,50 @@ func (b *Botanist) rotateKubeconfigSecrets(ctx context.Context, gardenerResource
 	return b.K8sGardenClient.Client().Patch(ctx, b.Shoot.Info, client.MergeFrom(oldObj))
 }
 
+func (b *Botanist) rotateSSHKeypairSecrets(ctx context.Context, gardenerResourceDataList *gardencorev1alpha1helper.GardenerResourceDataList) error {
+	var err error
+
+	// fetch the current SSH keypair
+	currentSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.SecretNameSSHKeyPair, Namespace: b.Shoot.SeedNamespace}}
+
+	if err = b.K8sSeedClient.Client().Get(ctx, client.ObjectKeyFromObject(currentSecret), currentSecret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get SSH keypair secret: %v", err)
+	}
```

Comment:
I am wondering why you are not getting the secret from the botanist? I mean something like:
```suggestion
	currentSecret, ok := b.Secrets[v1beta1constants.SecretNameSSHKeyPair]

	if !ok {
		return fmt.Errorf("failed to get SSH keypair secret: %v", err)
	}
```

Reply:
There was no special reason, I was simply not super familiarwith how the `Secrets` map is constructred and used during the botanisting phase. Your suggestion is of course the right approach.